### DISCUSSION TITLE
How to monitor specific user message?

### DISCUSSION THREAD

**Discussion by lieyongchang** [telegraf/telegraf#1571] [Q&A]
As the title suggest.

below code is what I have so far

```
const { Telegraf } = require('telegraf')
const bot = new Telegraf(processes.env.BOT_TOKEN);


bot.command('start', ctx => {
  console.log(ctx.from)
  bot.telegram.sendMessage(ctx.chat.id, 'hello there! Welcome to my new telegram bot.', {
  })
})
```

The above code only execute when I start my bot, but that is not what I want. I want my bot to listen from user A, and trigger an action when 'hear' certain text 

any advice is appreciated

**Comment by kraykin93** [comment]
You want to use `hears` middleware.

```
const { Telegraf } = require('telegraf')

const bot = new Telegraf(process.env.BOT_TOKEN)
bot.hears('hi', (ctx) => ctx.reply('Hey there'), { reply_to_message_id: ctx.message.message_id })
bot.launch()
```

But it watches only for `text`. I personally look for something that watches for `caption` too.
